scalafix-tests/unit/src/test/scala/scalafix/tests/core/PatchSuite.scala

```diff
@@ -178,4 +179,46 @@ class PatchSuite extends AbstractSyntacticRuleSuite with AnyFunSuiteLike {
         fail(s"$atomicPatch is not of expected type AtomicPatch(Concat(_, _))")
     }
   }
+
+  case class ReplaceTreeRule(pf: PartialFunction[Tree, String])
+      extends SyntacticRule("replaceTreeRule") {
+
+    override def fix(implicit doc: SyntacticDocument): Patch = {
+      val patch = doc.tree.collect {
+        case t if pf.isDefinedAt(t) => Patch.replaceTree(t, pf(t))
+      }.asPatch
+      assert(patch.nonEmpty)
+      patch
+    }
+  }
+
+  val program: Source =
+    """import a.b
+      |
+      |class Foo(private val q: String = "") extends Bar with Qux {
```

Comment:
`class Foo extends Bar` was enough to reproduce the problem but I thought I might as well write a longer program to prevent similar "symmetry" regressions for other trees, but it's of course by no mean exhaustive... I wonder if there is a test input somewhere in scalameta that makes use of all tree types?